internal/operators/ocs/ocs_operator_test.go

```diff
@@ -0,0 +1,142 @@
+			),
+			table.Entry("there are 3 hosts, role of one as auto-assign",
+				&common.Cluster{Cluster: models.Cluster{Hosts: []*models.Host{
+					master1, master2, autoAssignHost,
+				}}},
+				autoAssignHost,
+				&models.ClusterHostRequirementsDetails{CPUCores: CPUCompactMode + operator.config.OCSRequiredDiskCPUCount, RAMMib: conversions.GbToMib(MemoryGBCompactMode + operator.config.OCSRequiredDiskRAMGB), DiskSizeGb: MinDiskSize},
+			),
+			table.Entry("there are 6 hosts, master requirements",
+				&common.Cluster{Cluster: models.Cluster{Hosts: []*models.Host{
+					master1, master2, master3, worker1, worker2, worker3,
+				}}},
+				master1,
+				&models.ClusterHostRequirementsDetails{CPUCores: 0, RAMMib: 0},
+			),
+			table.Entry("there are 6 hosts, worker with two disk requirements",
+				&common.Cluster{Cluster: models.Cluster{Hosts: []*models.Host{
+					master1, master2, master3, worker1, worker2, worker3,
+				}}},
+				worker1,
+				&models.ClusterHostRequirementsDetails{CPUCores: CPUMinimalMode + operator.config.OCSRequiredDiskCPUCount, RAMMib: conversions.GbToMib(MemoryGBMinimalMode + operator.config.OCSRequiredDiskRAMGB), DiskSizeGb: MinDiskSize},
+			),
+			table.Entry("there are 6 hosts, worker with three disk requirements",
+				&common.Cluster{Cluster: models.Cluster{Hosts: []*models.Host{
+					master1, master2, master3, worker1, worker2, worker3,
+				}}},
+				worker2,
+				&models.ClusterHostRequirementsDetails{CPUCores: CPUMinimalMode + 2*operator.config.OCSRequiredDiskCPUCount, RAMMib: conversions.GbToMib(MemoryGBMinimalMode + 2*operator.config.OCSRequiredDiskRAMGB), DiskSizeGb: MinDiskSize},
+			),
+			table.Entry("there are 6 hosts, worker with one disk requirements",
+				&common.Cluster{Cluster: models.Cluster{Hosts: []*models.Host{
+					master1, master2, master3, worker1, worker2, worker3,
+				}}},
+				worker3,
+				&models.ClusterHostRequirementsDetails{CPUCores: CPUMinimalMode, RAMMib: conversions.GbToMib(MemoryGBMinimalMode)},
+			),
+		)
+		table.DescribeTable("failure scenario: get requirements for hosts when ", func(cluster *common.Cluster, host *models.Host, expectedResult error) {
+			_, err := operator.GetHostRequirements(ctx, cluster, host)
+			Expect(err).To(BeEquivalentTo(expectedResult))
+		},
+			table.Entry("Single master",
+				&common.Cluster{Cluster: models.Cluster{Hosts: []*models.Host{
+					master1,
+				}}},
+				master1,
+				fmt.Errorf("OCS require a minimum of 3 hosts"),
+			),
+			table.Entry("there are two master",
+				&common.Cluster{Cluster: models.Cluster{Hosts: []*models.Host{
+					master1, master2,
+				}}},
+				master1,
+				fmt.Errorf("OCS require a minimum of 3 hosts"),
+			),
+			table.Entry("no disk in one of the master",
+				&common.Cluster{Cluster: models.Cluster{Hosts: []*models.Host{
+					master1, master2, master3,
+				}}},
+				master2,
+				fmt.Errorf("OCS requires a minimum of 1 disk per host"),
+			),
```

Comment:
Also add a test scenario with 1 of the master having only 1 bootable disk.

Reply:
done